src/core/net/ip6.cpp

```diff
@@ -1010,11 +1010,13 @@ otError Ip6::ProcessReceiveCallback(Message &          aMessage,
     if (mIsReceiveIp6FilterEnabled)
     {
         // do not pass messages sent to an RLOC/ALOC, except Service Locator
+#if !OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
```

Comment:
Explain: we still need to pass the messages with ALOC/RLOC destinations to the tun interface when PLAT_UDP is used. Otherwise, UDP servers can not receive any message when PLAT_UDP is enabled.